Source/WebKit/GPUProcess/graphics/RemoteGraphicsContextGL.cpp

```diff
@@ -219,7 +219,9 @@ void RemoteGraphicsContextGL::paintRenderingResultsToCanvas(WebCore::RenderingRe
 void RemoteGraphicsContextGL::paintRenderingResultsToCanvasWithQualifiedIdentifier(QualifiedRenderingResourceIdentifier imageBuffer, CompletionHandler<void()>&& completionHandler)
 {
     assertIsCurrent(workQueue());
-    paintPixelBufferToImageBuffer(m_context->readRenderingResultsForPainting(), imageBuffer, WTFMove(completionHandler));
+    m_context->withDrawingBufferAsPlatformImage([&](PlatformImagePtr&& image) {
+        paintPlatformImageToImageBuffer(WTFMove(image), imageBuffer, WTFMove(completionHandler));
```

Comment:
So here you could add:

```
// We know that the image might be updated afterwards, so flush the drawing so that read back does not
occur.
imageBuffer->flushDrawingContext();
```

Reply:
This flushDrawingContext is happening inside paintPlatformImageToImageBuffer.